presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentProvider.java

```diff
@@ -46,20 +45,14 @@ protected EnvironmentProvider(List<EnvironmentExtender> bases)
                 .add(environmentConfig)
                 .build();
 
-        compose(extenders).extendEnvironment(builder);
-        return builder;
-    }
-
-    static EnvironmentExtender compose(Iterable<EnvironmentExtender> extenders)
-    {
-        String extendersNames = StreamSupport.stream(extenders.spliterator(), false)
+        String extendersNames = extenders.stream()
                 .map(Object::getClass)
                 .map(Class::getSimpleName)
                 .collect(joining(", "));
 
-        return builder -> {
-            log.info("Building environment %s with extenders: %s", builder.getEnvironmentName(), extendersNames);
-            extenders.forEach(extender -> extender.extendEnvironment(builder));
-        };
+        log.info("Building environment %s with extenders: %s", builder.getEnvironmentName(), extendersNames);
```

Comment:
The sense of previous code was to log message lazy (when environment is built, ot when compose is called)